selfdrive/car/toyota/carstate.py

```diff
@@ -163,6 +165,27 @@ def update(self, cp, cp_cam):
     if self.CP.carFingerprint != CAR.PRIUS_V:
       self.lkas_hud = copy.copy(cp_cam.vl["LKAS_HUD"])
 
+    # Convert distance lines (3, 2, 1) to personality profiles (2, 1, 0)
+    personality = cp.vl["PCM_CRUISE_SM"]["DISTANCE_LINES"] - 1
+
+    # Restore the previous drive's personality to the dash
+    if personality == self.previous_personality:
+      self.profile_restored = True
+    elif not self.profile_restored:
+      # Can send a distance pressed command every other frame to quickly restore the previous drive's personality profile
+      self.distance_button = not self.distance_button
+
+    elif self.CP.flags & ToyotaFlags.SMART_DSU:
+      self.distance_button = cp.vl["SDSU"]["FD_BUTTON"] == 1
+    # This may need some additional checks or can even simply be an "else" statement if all cars already have it mapped out
+    elif self.CP.openpilotLongitudinalControl:
+      self.distance_button = cp_acc.vl["ACC_CONTROL"]["DISTANCE"] == 1
```

Comment:
Note that we disable the radar on some TSS2 cars

Reply:
Yeah that’s one of the additional checks I was thinking I’ll need to add in. I’ll keep that in mind when I go through and clean it up tomorrow.

Reply:
Apparently the buttons are wired directly up to the camera. I blocked all messages that we normally forward through the panda and `ACC_CONTROL` still knew when we were pressing both the distance and lane keep button :(

Looks like no button parsing for RAV4 2022+